### DISCUSSION TITLE
yolov8 removing labels and boxes and having pedestrian count

### DISCUSSION THREAD

**Discussion by mannaei77** [ultralytics/ultralytics#2283] [General]
greetings 
I have been having issues with the code below ,i need it to show the pedestrian count and remove the boxes and labels with no success we have trained ultralytics data set on  visdrone using yolov8x and was trained on local machine, part two of the question is how to increase the accuracy during training. I have epoch set to 100 if we increase it to 200 or 500 will i get better trained model.i am using thonny IDE
please Advise  
import cv2
import cvlib as cv
import pandas as pd
import numpy as np
from vidgear.gears import CamGear
from cvlib.object_detection import draw_bbox
from ultralytics import YOLO

model = YOLO('visdrone_last.pt')

stream = CamGear(source='https://www.youtube.com/watch?v=RQA5RcIZlAM', stream_mode=True, logging=True).start() # YouTube Video URL Video URL as input

my_file = open("coco.txt", "r")
data = my_file.read()
class_list = data.split("\n")
print(class_list)

count = 0
while True:
    
    frame = stream.read()
    count += 1
    if count % 5 != 0:
        continue
    
    frame = cv2.resize(frame, (1020, 600))
    bbox, label, conf = cv.detect_common_objects(frame)
    frame = draw_bbox(frame, bbox, label, conf)

    # Object counting
    car_count = label.count('car')
    pedestrian_count = label.count('pedestrian')
    person_count = label.count('person')
    truck_count = label.count('truck')
    m_cycle_count = label.count('motorcycle')

    cv2.putText(frame, f'Cars: {car_count}', (30, 480), cv2.FONT_HERSHEY_PLAIN, 2, (255, 255, 255), 2)
    cv2.putText(frame, f'People: {person_count}', (30, 510), cv2.FONT_HERSHEY_PLAIN, 2, (255, 255, 255), 2)
    cv2.putText(frame, f'Trucks: {truck_count}', (30, 544), cv2.FONT_HERSHEY_PLAIN, 2, (255, 255, 255), 2)
    cv2.putText(frame, f'Motorcycles: {m_cycle_count}', (30, 574), cv2.FONT_HERSHEY_PLAIN, 2, (255, 255, 255), 2)
    cv2.putText(frame, f'Pedestrians: {pedestrian_count}', (30, 400), cv2.FONT_HERSHEY_PLAIN, 2, (255, 255, 255), 2)

    results = model.predict(frame, show=True)
           
    if cv2.waitKey(1) & 0xFF == 27:
        break

stream.release()
cv2.destroyAllWindows()

**Comment by glenn-jocher** [comment]
@mannaei77 hello! To count pedestrians and remove the bounding boxes and labels, you can modify your code to use the predictions from the YOLOv8 model without drawing the boxes. You can access the predictions directly from the model's output and count the instances of the 'pedestrian' class.

Regarding improving accuracy during training, increasing the number of epochs can potentially lead to a better model, but it's not guaranteed. It's important to monitor the validation loss to ensure that the model is not overfitting. Additionally, consider experimenting with different hyperparameters, augmentations, and using a more diverse dataset. Make sure to refer to the [Train](https://docs.ultralytics.com/modes/train) documentation for more detailed guidance on training your model effectively.

For the pedestrian count without drawing boxes, you can use the `model.predict()` method to get the predictions and then count the occurrences of the 'pedestrian' class in the labels. You would skip the `draw_bbox` function entirely.

Remember to check the [Predict](https://docs.ultralytics.com/modes/predict) documentation for more information on how to use the model for predictions without drawing the bounding boxes.

Keep in mind that for specific help with code, you might want to refer to the [Ultralytics Docs](https://docs.ultralytics.com) or seek community support on the GitHub discussions for the repo. Good luck with your project! 😊
